packages/deployments/contracts/contracts/messaging/libraries/MerkleLib.sol

```diff
@@ -104,36 +105,734 @@ library MerkleLib {
     revert MerkleLib__insert_treeIsFull();
   }
 
+  // ========= Read Methods =========
+
   /**
    * @notice Calculates and returns tree's current root.
-   * @return bytes32 root.
+   * @return _current bytes32 root.
    **/
-  function root(Tree storage tree) internal view returns (bytes32) {
-    return rootWithCtx(tree, zeroHashes());
-  }
+  function root(Tree storage tree) internal view returns (bytes32 _current) {
+    uint256 _index = tree.count;
 
-  // ========= Helper Methods =========
+    if (_index == 0) {
+      return Z_32;
+    }
 
-  /**
-   * @notice Calculates and returns tree's current root given array of zero hashes.
-   * @param _zeroes Array of zero hashes.
-   * @return _current Calculated root of tree.
-   **/
-  function rootWithCtx(Tree storage tree, bytes32[TREE_DEPTH] memory _zeroes) internal view returns (bytes32 _current) {
-    uint256 _index = tree.count;
+    uint256 i;
+    assembly {
```

Comment:
So this should be replaced with the original Solidity code  (assuming YUL won't be used)

Reply:
@gpersoon I'm a bit confused. Why would we not use the YUL / assembly code here? We want to use it. Assuming you are referring to the `root` method here, we are calling it in the codebase. Sry, but if you could clarify what you mean here that would be great!

Reply:
Hi Jake,
I was referring to: https://github.com/spearbit-audits/connext-nxtp/issues/167#issuecomment-1317415966
"We don't have the internal capability to maintain Yul safely, so adding a large portion to core code paths seems inadvisable"
So that is why I was confused :-(